### ISSUE TITLE
Predictions output file format and naming

### ISSUE THREAD

**Issue by youngdjn** [open-forest-observatory/tree-detection-framework#43]
Currently, `RegionDetections.save` saves an ESRI shapefile of name `name.shp` (plus ancillary files) into a folder named `name`, where `name` is provided by the user via the argument `save_path`, or the entrypoint arg `--predictions-save-path`.

It would be better to default to `.gpkg` output file format ([see here](http://switchfromshapefile.org/)). Also, I think it would be best for the output filename to be the same as the input raster used for predictions (probably with a datetime suffix, so multiple preds can be made without overwriting). This way, if they user specifies to perform predictions for multiple rasters (e.g. the input folder contains multiple rasters), predictions from each one can be saved and associated with their source.

**Comment by russelldj** [comment]
I agree that `shp` is a bad format to use and it wasn't my goal to use it initially. It looks like if you the path you provide to `geopandas.save_file` doesn't have an extension, it interprets it as `shp`. I think there's a couple documentation changes that could make this a little better. The first is already done, since @amrithasp02 clarified the output file should be a geofile [here](https://github.com/open-forest-observatory/tree-detection-framework/blob/3a4d747a7524d421795460a4f3d1ae63e8b58f62/tree_detection_framework/entrypoints/generate_predictions.py#L55). I can also update the documentation of `RegionDetections.save` ([here](https://github.com/open-forest-observatory/tree-detection-framework/blob/3a4d747a7524d421795460a4f3d1ae63e8b58f62/tree_detection_framework/detection/region_detections.py#L180)) to be a little more explicit that a file path with a geospatial extension is requested.

I see the reasoning for per-raster outputs, but that was not the way I initially thought of this working. I was thinking we'd provide a folder of rasters only if they were overlapping or otherwise needed to be processed together. In this case, I don't think we'd necessarily want per-raster outputs, and for the overlapping case, this might not even be clearly defined. But I can see an alternative workflow where you just want to conveniently run on a folder of independent rasters.

If we do want to do some sort of default naming, I think the [generate_predictions](https://github.com/open-forest-observatory/tree-detection-framework/blob/3a4d747a7524d421795460a4f3d1ae63e8b58f62/tree_detection_framework/entrypoints/generate_predictions.py#L14) function would be a good place for that logic.

**Comment by youngdjn** [comment]
Oh thanks, I think I misunderstood a few things before, but everything you say makes sense and I don't think there's a need for any change, except maybe the documentation change you suggest (so I won't close this issue yet).
